src/Microsoft.Diagnostics.Monitoring.WebApi/Metrics/MetricsService.cs

```diff
@@ -58,6 +58,7 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
                     using IDisposable monitorListener = _optionsMonitor.OnChange((_, _) => optionsTokenSource.SafeCancel());
 
                     MetricsPipelineSettings counterSettings = MetricsSettingsFactory.CreateSettings(counterOptions, options);
+                    counterSettings.UseSharedSession = pi.EndpointInfo.RuntimeVersion.Major >= 8;
```

Comment:
This feels a little clunky by not wiring it through `CreateSettings` (same with MetricsOperationFactory), but if we make that method responsible for knowing the runtime version that'll impact a lot of callers. What I'm doing here (I think) is the most compact option that pushes the logic into `dotnet-monitor`, but then we need to manually set `UseSharedSession` on settings, which I guess I'm not sure if that's something we care about.